trace/trace.go

```diff
@@ -178,25 +169,23 @@ func (t *Trace) logWithStepThreshold(stepThreshold time.Duration) {
 
 func writeTrace(b *bytes.Buffer, t *Trace, formatter string, stepThreshold time.Duration) time.Time {
 	lastStepTime := t.startTime
-	stepAndTraces := t.sortStepTrace()
+	stepAndTraces := t.stepsTraces
 	if len(stepAndTraces) == 0 {
 		return lastStepTime
 	}
 	for _, stepOrTrace := range stepAndTraces {
-		blankTime := time.Time{}
-		stepTime := stepOrTrace.writeStep(b, formatter, stepThreshold, lastStepTime)
-		if stepTime != blankTime {
-			lastStepTime = stepTime
-		}
+		stepTime := stepOrTrace.writeStep(b, formatter, t, stepThreshold, lastStepTime)
+		lastStepTime = stepTime
 	}
 	return lastStepTime
 }
 
 // LogIfLong is used to dump steps that took longer than its share
 func (t *Trace) LogIfLong(threshold time.Duration) {
+	t.threshold = &threshold
 	if time.Since(t.startTime) >= threshold {
 		// if any step took more than it's share of the total allowed time, it deserves a higher log level
-		stepThreshold := threshold / time.Duration(len(t.steps)+1)
+		stepThreshold := threshold / time.Duration(len(t.stepsTraces)+1)
 		t.logWithStepThreshold(stepThreshold)
```

Comment:
I wasn't expecting nested traces to get logged. https://github.com/kubernetes/utils/pull/139#discussion_r405783089

Reply:
So we should use only steps for calculating the stepThreshold? Should I ping you on slack?